src/storage/serializer/json/JSON.cc

```diff
@@ -8,15 +8,19 @@ namespace zeek::storage::serializer::json {
 
 std::unique_ptr<Serializer> JSON::Instantiate() { return std::make_unique<JSON>(); }
 
-JSON::JSON() : Serializer("JSON") {}
+JSON::JSON() : Serializer("JSON") { versioned_name = util::fmt("%sv%d", name.c_str(), Val::JSON_VERSION); }
 
 std::optional<util::byte_buffer> JSON::Serialize(ValPtr val) {
+    static auto byte_converter = [](u_char c) { return std::byte(c); };
+
+    std::string version = versioned_name + ";";
```

Comment:
The `;` was my initial thought. I'm now thinking whether of `\n` or `<space>` might be easier to work with (e.g. using default `split()` helpers). Mind making the call?

Reply:
It doesn't matter to me. I'm not sure about using `util::split` though because we only want to split at the first instance and not potentially any other occurrences of the delimiter. `util::split` splits all of them.